File: MemoryVinyl/Core/Models/Song.swift
import Foundation

struct Song: Identifiable, Codable, Hashable {
    let id: UUID
    let songName: String
    let artist: String
    let lyricLines: [String]
    let emotionTags: [String]
    let reason: String

    init(
        id: UUID = UUID(),
        songName: String,
        artist: String,
        lyricLines: [String],
        emotionTags: [String],
        reason: String
    ) {
        self.id = id
        self.songName = songName
        self.artist = artist
        self.lyricLines = lyricLines
        self.emotionTags = emotionTags
        self.reason = reason
    }
}

File: MemoryVinylAndroid/app/src/main/java/com/memoryvinyl/model/Song.kt
package com.memoryvinyl.model

import java.util.UUID

data class Song(
    val id: String = UUID.randomUUID().toString(),
    val songName: String,
    val artist: String,
    val lyricLines: List<String>,
    val emotionTags: List<String>,
    val reason: String
)
